catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/content/data/impl/ContentItemImpl.java

```diff
@@ -89,8 +105,25 @@ public ContentItemImpl(String id, ByteSource byteSource, String mimeTypeRawData,
      */
     public ContentItemImpl(String id, ByteSource byteSource, String mimeTypeRawData,
             String filename, long size, Metacard metacard) {
+        this(id, "", byteSource, mimeTypeRawData, filename, size, metacard);
+    }
+
+    /**
+     * An incoming content item where the item's GUID and size should be known.
+     *
+     * @param id              the {@link ContentItem}'s GUID - can be null
+     * @param qualifier       the {@link ContentItem}'s qualifier - can be null
+     * @param byteSource      the {@link ContentItem}'s input stream containing its actual data
+     * @param mimeTypeRawData the {@link ContentItem}'s mime type
```

Comment:
Added validation to prevent these illegal characters in the qualifier & filename.